message_ix_models/testing/__init__.py

```diff
@@ -351,3 +353,72 @@ def not_ci(reason=None, action="skip"):
     """
     action = "skipif" if action == "skip" else action
     return getattr(pytest.mark, action)(condition=GHA, reason=reason)
+
+
+def unpack_snapshot_data(context: Context, snapshot_id: int):
+    """Already-unpacked data for a snapshot.
+
+    This copies the .csv.gz files from message_ix_models/data/test/… to the directory
+    where they *would* be unpacked by .model.snapshot._unpack. This causes the code to
+    skip unpacking them, which can be very slow.
+    """
+    if snapshot_id != 0 and snapshot_id != 1:
+        log.info(f"No unpacked data for snapshot {snapshot_id}")
+        return
+
+    dest = context.get_cache_path(
+        f"snapshot-{snapshot_id}",
+        "MESSAGEix-GLOBIOM_1.1_R11_no-policy_baseline",
+    )
+    log.debug(f"{dest = }")
+
+    snapshot_data_path = util.package_data_path(
+        "test",
+        f"snapshot-{snapshot_id}",
+        "MESSAGEix-GLOBIOM_1.1_R11_no-policy_baseline",
+    )
+    log.debug(f"{snapshot_data_path = }")
+
+    shutil.copytree(snapshot_data_path, dest, dirs_exist_ok=True)
+
+
+@pytest.fixture(
+    scope="session",
+    params=[
+        int(k.split("-")[1]) for k in util.pooch.SOURCE if k.startswith("snapshot")
+    ],
+)
+def load_snapshot(request, session_context, solved: bool = False):
```

Comment:
The pytest style (that we try to follow) seems to be that fixture names are nouns instead of verbs like ordinary function names: for example, `tmp_path` instead of `make_tmp_path`.

So here perhaps `loaded_snapshot`?

We can also type annotate the function:
```py
def load_snapshot(...) -> "Scenario":
```
or similar.

Reply:
The type hint apparently needs to be `Generator[message_ix.Scenario, None, None]` based on [this answer](https://stackoverflow.com/a/38423388/23037943) since we're using `yield` instead of `return` in the fixtures. Hope that's fine.